src/lib/gundb/UserStorageClass.js

```diff
@@ -1660,67 +1662,82 @@ export class UserStorage {
    * @returns {Promise} Promise with an array of feed events
    */
   async getFeedPage(numResults: number, reset?: boolean = false): Promise<Array<FeedEvent>> {
+    let { feed, feedIndex, cursor, feedIds, wallet } = this
+
     if (reset) {
-      this.cursor = undefined
-    }
-    if (this.cursor === undefined) {
-      this.cursor = 0
+      cursor = 0
     }
-    let total = 0
-    if (!this.feedIndex) {
+
+    if (!feedIndex) {
       return []
     }
 
-    let daysToTake: Array<[string, number]> = takeWhile(this.feedIndex.slice(this.cursor), day => {
-      if (total >= numResults) {
-        return false
+    // running through the days history until we got the request numResults
+    // storing days selected to the daysToTake
+    let total = 0
+    let daysToTake = takeWhile(feedIndex.slice(cursor), ([, eventsAmount]) => {
+      const takeDay = total < numResults
+
+      if (takeDay) {
+        total += eventsAmount
       }
-      total += day[1]
-      return true
+
+      return takeDay
     })
-    this.cursor += daysToTake.length
 
-    //TODO: WTF does this work?!?! if we JSON.stringify teh day index content how come we don't need to JSON.parse it?
-    //this works in the unit tests also
-    let promises: Array<Promise<Array<FeedEvent>>> = daysToTake.map(day => {
-      return this.feed
-        .get(day[0])
+    cursor += daysToTake.length
+    assign(this, { cursor })
```

Comment:
this is cryptic voodoo. the idiomatic way to do that is well simply this.cursor +=

Reply:
fixed